### DISCUSSION TITLE
Managing potential mentee expectations

### DISCUSSION THREAD

**Discussion by beeme1mr** [cncf/mentoring#876] [Google Summer of Code 2023]
Hi! I've signed up as a mentor of a GSoC project. The proposal has led to a significate amount of interest from students. That's obviously a great problem to have! However, if they're not selected, I'm concerned that the inherent competition may lead to a poor first time experience for people new to open source. How have others handled this in the past? I want to make it clear to the students how they can improve their odds of being selected without implying that it will guarantee their selection.

**Comment by aliok** [comment]
hi @beeme1mr . 

> That's obviously a great problem to have!

It really is!

You can tell them that they can improve their odds by:
- Interacting with the community: join chat channels / mailing lists, introduce themselves, ask questions (good questions), help others (for example, help candidates who had the same issue with you while going through a tutorial)
- Trying out the project: try on local environment, go through the tutorials/demos/documentation, file issues in case of problems, ask questions when stuck
- Trying to make some code contribution: try to fix some `good first issue`s
- Preparing a project proposal sooner rather than later for getting feedback (GSoC candidates will need to submit a [proposal](https://google.github.io/gsocguides/student/writing-a-proposal))

Obviously, none of this is news. In the end, you as a mentor would want to have mentees who are capable and who you want to work with.

I would tell candidates that the best will be selected for the program. They would understand. However, GSoC is not the only place to contribute to your community. All of the candidates can simply contribute outside of GSoC and it will be still very beneficial for them. 

I also understand having the GSoC etiquette on a CV and getting some stipend are very good motivation. For that, CNCF participates in additional mentorship programs like LFX and Outreachy. Anybody who is not selected can continue engaging with the community and the project and they can try these other programs (That would imply your community participates in those programs too. Let's be careful there).

I said that the best will be selected for the program, but there is also a competition among all CNCF projects because Google often doesn't give the slots that the organizations requested.

> How have others handled this in the past?

I personally haven't gone into the discussion of the competition aspect of the program with the candidates. I would rather give general advices to contribute, like I wrote above, and leave it there.

I hope this helps a little.

Thanks for bringing this up, great discussion!
